internal/daemon/daemon.go

```diff
@@ -550,6 +565,17 @@ func (d *Daemon) StartAPI(bootstrap bool, initConfig map[string]string, newConfi
 		return err
 	}
 
+	// Start all extension servers before post-join hook.
+	for _, extensionServer := range d.extensionServers {
+		server := d.initServer(extensionServer.Resources...)
+		url := api.NewURL().Scheme(extensionServer.Protocol).Host(extensionServer.Address.String())
+		network := endpoints.NewNetwork(d.shutdownCtx, endpoints.EndpointNetwork, server, *url, extensionServer.Certificate)
+		err = d.endpoints.Add(network)
```

Comment:
FWIW, this function is variadic so you can aggregate each item returned from `NewNetwork` and run `d.endpoints.Add(networks...)`.